godtools/App/Flows/LessonFlow.swift

```diff
@@ -134,7 +134,15 @@ class LessonFlow: NSObject, ToolNavigationFlow, Flow {
             }
         
         case .closeTappedFromLesson(let lesson, let highestPageNumberViewed):
-            flowDelegate?.navigate(step: .lessonFlowCompleted(state: .userClosedLesson(lesson: lesson, highestPageNumberViewed: highestPageNumberViewed)))
+            
+            let stackCount: Int = navigationController.viewControllers.count
+            
+            if stackCount == 1 {
+                flowDelegate?.navigate(step: .lessonFlowCompleted(state: .userClosedLesson(lesson: lesson, highestPageNumberViewed: highestPageNumberViewed)))
```

Comment:
interesting interaction between requirements I hadn't even though of, what do we do with showing the lesson feedback survey if the user jumps to a different lesson.

Reply:
Hey @frett that's a good question.  At this point survey will only show when leaving the initial lesson.